system/indy-node-tests/test_vc.py

```diff
@@ -12,45 +12,42 @@
 # logger = logging.getLogger(__name__)
 # logging.basicConfig(level=0, format='%(asctime)s %(message)s')
 
+
 @pytest.fixture(scope='function', autouse=True)
 @async_generator
 async def docker_setup_and_teardown(nodes_num):
     await setup_and_teardown(nodes_num)
 
 
 @pytest.mark.asyncio
-async def test_vc_by_restart(pool_handler, wallet_handler, get_default_trustee):
+async def test_vc_by_restart(
+        pool_handler, wallet_handler, get_default_trustee, nodes_num
+):
     trustee_did, _ = get_default_trustee
-    did1, _ = await did.create_and_store_my_did(wallet_handler, '{}')
-    did2, _ = await did.create_and_store_my_did(wallet_handler, '{}')
-    await send_and_get_nym(pool_handler, wallet_handler, trustee_did, did1)
+    await ensure_pool_is_functional(pool_handler, wallet_handler, trustee_did)
     primary_before, _, _ = await get_primary(pool_handler, wallet_handler, trustee_did)
-    print('\nPrimary before: {}'.format(primary_before))
     p1 = NodeHost(primary_before)
     p1.stop_service()
-    primary_after = await wait_until_vc_is_done(primary_before, pool_handler, wallet_handler, trustee_did)
-    print('\nPrimary after: {}'.format(primary_after))
-    assert primary_before != primary_after
-    await send_and_get_nym(pool_handler, wallet_handler, trustee_did, did2)
+    await ensure_primary_changed(pool_handler, wallet_handler, trustee_did, primary_before)
+    await ensure_pool_is_functional(pool_handler, wallet_handler, trustee_did)
     p1.start_service()
-    await eventually_positive(check_pool_is_in_sync)
+    await ensure_pool_is_in_sync(nodes_num=nodes_num)
+    await ensure_pool_is_functional(pool_handler, wallet_handler, trustee_did)
 
 
 @pytest.mark.asyncio
-async def test_vc_by_demotion(pool_handler, wallet_handler, get_default_trustee):
+async def test_vc_by_demotion(
```

Comment:
You might consider to use `check_no_failures_fixture` here as well.

Reply:
Done.